Installation issue: aocl-dlp missing header numa.h

### Steps to reproduce the issue

Package aocl-dlp@5.2 needs numa.h from numactl package.  Since amd-aocl requires aocl-dlp, aocl will also not build.

```console
$ spack install aocl-dlp@5.2

  >> 1418    /tmp/jenkins/spack-stage/spack-stage-aocl-dlp-5.2-ztqsitj7ha3hxnlv
             xsqc6ac7jdd4iuzp/spack-src/bench/bench_gemm.cc:70:10: fatal error:
              numa.h: No such file or directory
     1419       70 | #include <numa.h>
     1420          |          ^~~~~~~~
...
```


### Error message

<details><summary>Error message</summary>

<pre>
numa.h: No such file or directory
</pre></details>


### Information on your system

* **Spack:** 1.2.0.dev0 (https://github.com/spack/spack/commit/558c7eb2716ec0ec208ae34bad19f9683c934f51)
* **Builtin repo:** /home/gragghia/magma_release_tests/spack-packages/repos/spack_repo/builtin (c141a53)
* **Python:** 3.9.21
* **Platform:** linux-rocky9-skylake_avx512

[spack-build-env.txt](https://github.com/user-attachments/files/25584510/spack-build-env.txt)
[spack-build-out.txt](https://github.com/user-attachments/files/25584511/spack-build-out.txt)

### Additional information

@amd-toolchain-support

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

--- a/repos/spack_repo/builtin/packages/aocl_dlp/package.py
+++ b/repos/spack_repo/builtin/packages/aocl_dlp/package.py
@@ -41,6 +41,7 @@ class AoclDlp(CMakePackage):
     depends_on("c", type="build")
     depends_on("cxx", type="build")
     depends_on("cmake@3.26:", type="build")
+    depends_on("numactl", when="+benchmarks")
 
     def cmake_args(self):
         spec = self.spec

